test/filters.js

```diff
@@ -211,7 +211,10 @@ function filterTest(label, filterPayload, afterInstall, filterChangesOne, filter
     engine.addProvider(filterProvider)
     engine.addProvider(blockProvider)
     engine.once('block', startTest)
-    engine.start()
+
+    setTimeout(() => {
+      engine.start()
+    }, 1)
```

Comment:
Sometimes the `startTest` would be fired, and would check the filter subprovider for recognizing new filters before the filter subprovider had received the `block` event, which it uses to identify filter changes.

This timeout allows the filter subprovider an opportunity to cache updates to return on subsequent requests.